common/src/test/kotlin/streams/service/sink/strategy/CUDIngestionStrategyTest.kt

```diff
@@ -0,0 +1,572 @@
+
+    @Test
+    fun `should delete nodes`() {
+        // given
+        val detachMarkers = listOf(1, 3, 8, 10)
+        val list = (1..10).map {
+            val labels = if (it % 2 == 0) listOf("Foo", "Bar") else listOf("Foo", "Bar", "Label")
+            val detach = it in detachMarkers
+            val properties = mapOf("foo" to "foo-value-$it", "id" to it)
+            val cudNode = CUDNode(op = CUDOperations.delete,
+                    labels = labels,
+                    ids = mapOf("key" to it),
+                    properties = properties,
+                    detach = detach)
+            StreamsSinkEntity(null, cudNode)
+        }
+
+        // when
+        val cudQueryStrategy = CUDIngestionStrategy()
+        val nodeEvents = cudQueryStrategy.mergeNodeEvents(list)
+        val nodeDeleteEvents = cudQueryStrategy.deleteNodeEvents(list)
+
+        val relationshipEvents = cudQueryStrategy.mergeRelationshipEvents(list)
+        val relationshipDeleteEvents = cudQueryStrategy.deleteRelationshipEvents(list)
+
+        // then
+        assertEquals(emptyList(), nodeEvents)
+        assertEquals(emptyList(), relationshipEvents)
+        assertEquals(emptyList(), relationshipDeleteEvents)
+
+        assertEquals(4, nodeDeleteEvents.size)
+        assertEquals(10, nodeDeleteEvents.map { it.events.size }.sum())
+        val deleteNodeFooBar = findEventByQuery("""
+                |${StreamsUtils.UNWIND}
+                |MATCH (n:Foo:Bar {key: event.${CUDIngestionStrategy.ID_KEY}.key})
+                |DELETE n
+            """.trimMargin(), nodeDeleteEvents)
+        assertEquals(3, deleteNodeFooBar.events.size)
+        assertNodeEventsContainsKey(deleteNodeFooBar, "key")
+        val deleteNodeFooBarDetach = findEventByQuery("""
+                |${StreamsUtils.UNWIND}
+                |MATCH (n:Foo:Bar {key: event.${CUDIngestionStrategy.ID_KEY}.key})
+                |DETACH DELETE n
+            """.trimMargin(), nodeDeleteEvents)
+        assertEquals(2, deleteNodeFooBarDetach.events.size)
+        assertNodeEventsContainsKey(deleteNodeFooBarDetach, "key")
+        val deleteNodeFooBarLabel = findEventByQuery("""
+                |${StreamsUtils.UNWIND}
+                |MATCH (n:Foo:Bar:Label {key: event.${CUDIngestionStrategy.ID_KEY}.key})
+                |DELETE n
+            """.trimMargin(), nodeDeleteEvents)
+        assertEquals(3, deleteNodeFooBarLabel.events.size)
+        assertNodeEventsContainsKey(deleteNodeFooBarLabel, "key")
+        val deleteNodeFooBarLabelDetach = findEventByQuery("""
+                |${StreamsUtils.UNWIND}
+                |MATCH (n:Foo:Bar:Label {key: event.${CUDIngestionStrategy.ID_KEY}.key})
+                |DETACH DELETE n
+            """.trimMargin(), nodeDeleteEvents)
+        assertEquals(2, deleteNodeFooBarLabelDetach.events.size)
+        assertNodeEventsContainsKey(deleteNodeFooBarLabelDetach, "key")
+    }
+
+    @Test
+    fun `should delete nodes with internal id reference`() {
```

Comment:
Functionality / behavior question.  Not sure I saw this spot in the code, but what happens if the message tries to delete something by an internal ID which obviously isn't an internal ID?   Like id="banana".

This goes back to the earlier point about debug messaging when the plugin has to drop certain records because they're malformed.

Reply:
This goes into #237